tool-openssl/rsa_test.cc

```diff
@@ -0,0 +1,287 @@
+  }
+
+  void RunCommandsAndCompareOutput(const std::string &tool_command, const std::string &openssl_command) {
+    int tool_result = system(tool_command.c_str());
+    ASSERT_EQ(tool_result, 0) << "AWS-LC tool command failed: " << tool_command;
+
+    int openssl_result = system(openssl_command.c_str());
+    ASSERT_EQ(openssl_result, 0) << "OpenSSL command failed: " << openssl_command;
+
+    std::ifstream tool_output(out_path_tool);
+    this->tool_output_str = std::string((std::istreambuf_iterator<char>(tool_output)), std::istreambuf_iterator<char>());
+    std::ifstream openssl_output(out_path_openssl);
+    this->openssl_output_str = std::string((std::istreambuf_iterator<char>(openssl_output)), std::istreambuf_iterator<char>());
+
+    std::cout << "AWS-LC tool output:" << std::endl << this->tool_output_str << std::endl;
+    std::cout << "OpenSSL output:" << std::endl << this->openssl_output_str << std::endl;
+  }
+
+
+  void TearDown() override {
+    if (tool_executable_path != nullptr && openssl_executable_path != nullptr) {
+      RemoveFile(in_path);
+      RemoveFile(out_path_tool);
+      RemoveFile(out_path_openssl);
+    }
+  }
+
+  char in_path[PATH_MAX];
+  char out_path_tool[PATH_MAX];
+  char out_path_openssl[PATH_MAX];
+  bssl::UniquePtr<RSA> rsa;
+  const char* tool_executable_path;
+  const char* openssl_executable_path;
+  std::string tool_output_str;
+  std::string openssl_output_str;
+};
+
+// Helper function to trim whitespace from both ends of a string to test RSA output
+static inline std::string &trim(std::string &s) {
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char ch) {
+      return !std::isspace(static_cast<unsigned char>(ch));
+  }));
+  s.erase(std::find_if(s.rbegin(), s.rend(), [](unsigned char ch) {
+      return !std::isspace(static_cast<unsigned char>(ch));
+  }).base(), s.end());
+  return s;
+}
+
+// Helper function to read file content into a string
+// Helper function to read file content into a string
+std::string ReadFileToString(const std::string& file_path) {
+  std::ifstream file_stream(file_path, std::ios::binary);
+  if (!file_stream) {
+    return "";
+  }
+  std::ostringstream buffer;
+  buffer << file_stream.rdbuf();
+  return buffer.str();
+}
+
+
+// RSA boundaries
+const std::string RSA_BEGIN = "-----BEGIN RSA PRIVATE KEY-----";
+const std::string RSA_END = "-----END RSA PRIVATE KEY-----";
+const std::string BEGIN = "-----BEGIN PRIVATE KEY-----";
+const std::string END = "-----END PRIVATE KEY-----";
+const std::string MODULUS = "Modulus=";
+
+bool CheckBoundaries(const std::string &content, const std::string &begin1, const std::string &end1, const std::string &begin2, const std::string &end2) {
+  return (content.compare(0, begin1.size(), begin1) == 0 && content.compare(content.size() - end1.size(), end1.size(), end1) == 0) ||
+         (content.compare(0, begin2.size(), begin2) == 0 && content.compare(content.size() - end2.size(), end2.size(), end2) == 0);
+}
```

Comment:
Nice comparison function :) It might be helpful to document that this is due to different versions of OpenSSL having different PEM outputs and which version this started diverging.